autopush/crypto_key.py

```diff
@@ -0,0 +1,82 @@
+"""Crypto-Key header parser and manager"""
+
+
+class CryptoKeyException(Exception):
+    """Invalid CryptoKey"""
+
+
+class CryptoKey(object):
+    """Parse the Crypto-Key header per
+    http://tools.ietf.org/html/draft-ietf-httpbis-encryption-encoding-00#section-4
+
+    """
+    _values_ = []
+
+    def __init__(self, header):
+        """Parse the Crypto-Key header
+
+        :param header: Header content string.
+
+        """
+        chunks = header.split(",")
+        if self._values_:
+            self._values_ = []
+        for chunk in chunks:
+            bits = chunk.split(";")
+            hash = {}
+            for bit in bits:
+                try:
+                    (key, value) = bit.split("=", 1)
+                except ValueError:
+                    raise CryptoKeyException("Invalid Crypto Key value")
+                hash[key.strip()] = value.strip(' "')
+            self._values_.append(hash)
+
+    def get_keyid(self, keyid):
+        """Return the Crypto-Key hash referred to by a given keyid.
+
+        For example, for a CryptoKey specified as:
+        Crypto-Key: keyid="apple";foo="fruit",keyid="gorp";bar="snake"
+
+        get_keyid("apple") would return a hash of
+        {"keyid": "apple", "foo":"fruit"}
+
+        :param keyid: The keyid to reference
+        :returns: hash of the matching key data or None
+
+        """
+        for val in self._values_:
+            if keyid == val.get('keyid'):
+                return val
+        return None
+
+    def get_label(self, label):
+        """Return the Crypto-Key value referred to by a given label.
+
+        For example, for a CryptoKey specified as:
+        Crypto-Key: keyid="apple";foo="fruit",keyid="gorp";bar="snake"
+
+        get_label("foo")
+        would return a value of "apple"
+
+        **NOTE** This presumes that "label" is unique. Otherwise it will
```

Comment:
Since these are generally formatted with Sphinx using RST, this should be a ..note directive.